crates/eval/src/eval.rs

```diff
@@ -45,32 +35,59 @@ impl ForkedEvm {
     ///
     /// The typed return of the parse, plus Foundry's RawCallResult struct.
     pub async fn fork_parse(
-        self: &mut ForkedEvm,
-        args: ForkParseArgs,
-    ) -> Result<ForkTypedReturn<parseCall>, ForkCallError> {
-        let ForkParseArgs {
-            rainlang_string,
-            deployer,
-        } = args;
-        let mut executor = self.build_executor();
+        &mut self,
+        rainlang_string: &str,
+        deployer: Address,
+    ) -> Result<parseReturn, ForkCallError> {
```

Comment:
which one is that? the call to parse? or the call to deploy/integrity check?
because there are 2 transactions/calls, well that made me to just return the `parseReturn` on success